### PR TITLE
Fix metrics `WriteBufferFromS3Bytes`, `WriteBufferFromS3Microseconds` and `WriteBufferFromS3RequestsErrors`

### PR DESCRIPTION
**Pull request by AVMusorin**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix metrics `WriteBufferFromS3Bytes`, `WriteBufferFromS3Microseconds` and `WriteBufferFromS3RequestsErrors`

Ref: https://github.com/ClickHouse/ClickHouse/pull/45188

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 3c9b5b9dc35fea0b677a12c55bb5bd978e66b8e4 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49930/3c9b5b9dc35fea0b677a12c55bb5bd978e66b8e4/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/IO/S3/copyS3File.cpp [hunk 1]
```diff
@@ -184,7 +188,7 @@ namespace
                     LOG_INFO(log, "Multipart upload failed with NO_SUCH_KEY error for Bucket: {}, Key: {}, Upload_id: {}, Parts: {}, will retry", dest_bucket, dest_key, multipart_upload_id, part_tags.size());
                     continue; /// will retry
                 }
-
+                ProfileEvents::increment(ProfileEvents::WriteBufferFromS3RequestsErrors, 1);
```
Comment (antonio2368) [inline src/IO/S3/copyS3File.cpp:191]: same in `createMultipartUpload` before exception?

File: src/IO/S3/copyS3File.cpp [hunk 2]
```diff
@@ -15,6 +15,10 @@
 
 namespace ProfileEvents
 {
+    extern const Event WriteBufferFromS3Bytes;
+    extern const Event WriteBufferFromS3Microseconds;
+    extern const Event WriteBufferFromS3RequestsErrors;
```
Comment (azat) [inline src/IO/S3/copyS3File.cpp:20]: This metrics should be added to a single part upload as well

**Comment by azat** [comment]
>Fix metrics WriteBufferFromS3Bytes, WriteBufferFromS3Microseconds and WriteBufferFromS3RequestsErrors for native S3 copy function

By the way this is not "native copy" (i.e. not a CopyObject request), but a regular copy via buffers, for native copy there is `S3CopyObject` and it should work

File: src/IO/S3/copyS3File.cpp (hunk 1 above)
Comment (AVMusorin) [reply src/IO/S3/copyS3File.cpp:194]: Yes. Added. Thanks!

File: src/IO/S3/copyS3File.cpp (hunk 2 above)
Comment (AVMusorin) [reply src/IO/S3/copyS3File.cpp:20]: Done. Thanks

**Review by antonio2368 — APPROVED** [review]

**Comment by AVMusorin** [comment]
Failed `00612_http_max_query_size` doesn't look related to current changes.

`00992_system_parts_race_condition_zookeeper_long` should be fixed in https://github.com/ClickHouse/ClickHouse/issues/43946

**Pull request merged**